h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstTfIdf.java

```diff
@@ -0,0 +1,153 @@
+        dfOutFrame.remove(dfOutFrame.numCols() - 1);
+        dfOutFrame.add(IDF_COL_NAME, idfValues);
+
+        // TF
+        TermFrequency tfTask = new TermFrequency();
+        Frame tfOutFrame = tfTask.compute(wordFrame);
+        
+        // Intermediate frame containing both TF and IDF values
+        // TODO: Avoid converting a string column to categorical
+        tfOutFrame.replace(1, tfOutFrame.vecs()[1].toCategoricalVec());
+        dfOutFrame.replace(0, dfOutFrame.vecs()[0].toCategoricalVec());
+        int[][] levelMaps = {
+                CategoricalWrappedVec.computeMap(tfOutFrame.vec(1).domain(), dfOutFrame.vec(0).domain())
+        };
+        Frame tfIdfIntermediate = Merge.merge(tfOutFrame, dfOutFrame, new int[]{ 1 }, new int[]{ 0 }, false, levelMaps);
+        // TODO: Convert columnt back to string
+        tfIdfIntermediate.replace(1, tfIdfIntermediate.vecs()[1].toStringVec());
+
+        // TF-IDF
+        int tfOutFrameColCnt = tfIdfIntermediate.numCols();
+        TfIdf tfIdfTask = new TfIdf(tfOutFrameColCnt - 2, tfOutFrameColCnt - 1);
+        Vec tfIdfValues = tfIdfTask.doAll(new byte[]{ Vec.T_NUM }, tfIdfIntermediate).outputFrame().anyVec();
+
+        // Construct final frame containing TF, IDF and TF-IDF values
+        tfIdfIntermediate.add(TF_IDF_COL_NAME, tfIdfValues);
+        tfIdfIntermediate._key = Key.make();
+
+        System.out.println(tfIdfIntermediate.toTwoDimTable().toString());
+        
+        return new ValFrame(tfIdfIntermediate);
+    }
+
+    @Override
+    public String str() {
+        return "tf-idf";
+    }
+
+    /**
+     * Final TF-IDF Map-Reduce task used to combine TF and IDF values together.
+     */
+    private static class TfIdf extends MRTask<TfIdf> {
+        
+        // IN
+        /**
+         * Index of a column containing Term Frequency values in the input frame of this task.
+         */
+        private final int _tfColIndex;
+        /**
+         * Index of a column containing Inverse Document Frequency values in the input frame of this task.
+         */
+        private final int _idfColIndex;
+        
+        private TfIdf(int tfColIndex, int idfColIndex) {
+            _tfColIndex = tfColIndex;
+            _idfColIndex = idfColIndex;
+        }
+
+        @Override
+        public void map(Chunk[] cs, NewChunk nc) {
+            Chunk tfValues = cs[_tfColIndex];
+            Chunk idfValues = cs[_idfColIndex];
+            
+            for (int row = 0; row < tfValues._len; row++) {
+                nc.addNum(tfIdf(tfValues.at8(row), idfValues.atd(row)));
+            }
+        }
+
+        /**
+         * Computes TF-IDF value from given TF and IDF values.
+         * 
+         * @param tf    TF value to be used to compute TF-IDF.
+         * @param idf   IDF value to be used to compute TF-IDF.
+         *
+         * @return TF-IDF value from given TF and IDF values.
+         */
+        private static double tfIdf(final long tf, final double idf) {
```

Comment:
unnecessary function

Reply:
This was there just for the code readability, but I guess in this case it is quite obvious and strictly defined to be a product. I removed it.